ICAFIX/hcp_fix_multi_run

```diff
@@ -419,428 +430,482 @@ log_Msg "PWD: $DIR"
+                log_Msg "Run interpreted MATLAB/Octave (${matlab_interpreter[*]}) with code..."
+                log_Msg "$matlab_code"
+                "${matlab_interpreter[@]}" <<<"$matlab_code"
+                echo
+                ;;
+        esac
+    fi # if ICAmethod == MELODIC
+
+    # At this point (following melodic or icasso), ${concatfmrihp}_vnts is no longer necessary
+    $FSLDIR/bin/imrm ${concatfmrihp}_vnts
+
+    ## ===========================================================================
+    ## End skip in case of SkipICA
+    ## ===========================================================================
 
-# At this point the concatenated VN'ed time series (both volume and CIFTI, following the "1st pass" VN) can be deleted
-log_Msg "Removing the concatenated VN'ed time series"
-"$FSLDIR/bin/imrm" "${ConcatNameNoExt}_hp${hp}_vnts"
-/bin/rm -f "${ConcatNameNoExt}_Atlas_hp${hp}_vn.dtseries.nii"
-# Alternatively, if the volume version is needed for any reason, need to at least rename it to avoid conflict with
-# same named file that will get created by running functionhighpassandvariancenormalize on the concatfmrihp file
-# during the "2nd pass" VN (below)
-#$FSLDIR/bin/immv ${ConcatNameNoExt}_hp${hp}_vnts ${ConcatNameNoExt}_hp${hp}_vnts0
-
-# Nor do we need the concatenated demeaned time series (either volume or CIFTI)
-log_Msg "Removing the concatenated demeaned time series"
-"$FSLDIR/bin/imrm" "${ConcatNameNoExt}_demean"
-/bin/rm -f "${ConcatNameNoExt}_Atlas_demean.dtseries.nii"
-
-if ((ConcatOnly)); then
-	exit
+else
+    log_Msg "Skipping ICA and proceeding directly to FIX. All the files generated by ICA have to be present!"
+
+    # Set the variables
+    ConcatFolder=`dirname ${ConcatName}`
+    ConcatNameNoExt=$($FSLDIR/bin/remove_ext $ConcatName)  # No extension, but still includes the directory path
+    concatfmri=`basename ${ConcatNameNoExt}`  # Directory path is now removed
+    concatfmrihp=${concatfmri}_hp${hp}
+
+    # Switch to concat folder
+    cd ${ConcatFolder}
+
+    # Was backup requested
+    if [ -n "$FIXBackup" ]; then
+
+        # Is FIXBackup an absolute path?
+        if [[ "$FIXBackup" = /* ]]; then
+            FIXBackupFolder="$FIXBackup"
+        else
+            FIXBackupFolder="$(pwd)/${concatfmrihp}.ica/$FIXBackup"
+        fi
+
+        # Does the FIXBackup folder exist?
+        if [ ! -d "$FIXBackupFolder" ]; then
+            mkdir -p "${FIXBackupFolder}"
+            log_Msg "Created a folder for FIX backup at: ${FIXBackupFolder}."
+        fi
+
+        # Back up previous fix solution
+        if [ -d "${concatfmrihp}.ica/fix" ]; then
+            mv "${concatfmrihp}.ica/fix"/* "${FIXBackupFolder}"
+            log_Msg "Previous fix solution found and moved to ${FIXBackupFolder}."
+        fi
+
+        # Zip and back up previous scene files
+        for SceneFile in $(ls *.scene); do
```

Comment:
```suggestion
        for SceneFile in *.scene; do
```
Globbing is whitespace-safe, use it without feeding it through `ls`.

Reply:
Also, if there is a substantial chance that there won't be any scene files, you should use `shopt -s nullglob` near the top of the file (like right under `set -eu`), to avoid "file not found: *.scene", but then you DEFINITELY need to avoid `ls`, as it does something completely different when passed zero arguments.

Reply:
@coalsont thank you for spotting this! It is now changed as suggested.